kanidmd/src/lib/core/https/mod.rs

```diff
@@ -1161,6 +262,139 @@ impl<State: Clone + Send + Sync + 'static> tide::listener::Listener<State> for T
+    async fn handle(
+        &self,
+        request: tide::Request<State>,
+        next: tide::Next<'_, State>,
+    ) -> tide::Result {
+        let mut response = next.run(request).await;
+        response.insert_header("Cache-Control", "max-age=60,must-revalidate,private");
+        Ok(response)
+    }
+}
+
+struct StaticContentMiddleware;
+
+impl Default for StaticContentMiddleware {
+    fn default() -> Self {
+        StaticContentMiddleware {}
+    }
+}
+
+#[async_trait::async_trait]
+impl<State: Clone + Send + Sync + 'static> tide::Middleware<State> for StaticContentMiddleware {
+    async fn handle(
+        &self,
+        request: tide::Request<State>,
+        next: tide::Next<'_, State>,
+    ) -> tide::Result {
+        let mut response = next.run(request).await;
+        response.insert_header("Cache-Control", "max-age=3600,private");
+        Ok(response)
+    }
+}
+
+struct StrictResponseMiddleware;
+
+impl Default for StrictResponseMiddleware {
+    fn default() -> Self {
+        StrictResponseMiddleware {}
+    }
+}
+
+#[async_trait::async_trait]
+impl<State: Clone + Send + Sync + 'static> tide::Middleware<State> for StrictResponseMiddleware {
+    async fn handle(
+        &self,
+        request: tide::Request<State>,
+        next: tide::Next<'_, State>,
+    ) -> tide::Result {
+        let mut response = next.run(request).await;
+        response.insert_header("x-frame-options", "deny");
+        response.insert_header("x-content-type-options", "nosniff");
+        response.insert_header("cross-origin-resource-policy", "same-origin");
+        response.insert_header("cross-origin-embedder-policy", "require-corp");
+        response.insert_header("cross-origin-opener-policy", "same-origin");
+        Ok(response)
+    }
+}
+
+struct StrictRequestMiddleware;
+
+impl Default for StrictRequestMiddleware {
+    fn default() -> Self {
+        StrictRequestMiddleware {}
+    }
+}
+
+#[async_trait::async_trait]
+impl<State: Clone + Send + Sync + 'static> tide::Middleware<State> for StrictRequestMiddleware {
+    async fn handle(
+        &self,
+        request: tide::Request<State>,
+        next: tide::Next<'_, State>,
+    ) -> tide::Result {
+        let proceed = {
+            match request.header("sec-fetch-site") {
+                None => true,
+                Some(hv) => {
+                    let hvs = hv.as_str();
+                    if hvs == "same-origin" || hvs == "same-site" || hvs == "none" {
+                        true
+                    } else {
+                        request.header("sec-fetch-mode").map(|v| v.as_str()) == Some("navigate")
+                            && request.method() == tide::http::Method::Get
+                            && request.header("sec-fetch-dest").map(|v| v.as_str())
+                                != Some("object")
+                            && request.header("sec-fetch-dest").map(|v| v.as_str()) != Some("embed")
```

Comment:
I think I'd rather there be `( ) ` around the components after the `||` just because I've been bitten by assuming operator ordering in the past. Otherwise I'd commit it (maybe I'm too old, I can't find how to edit your suggestion >.>)

Reply:
Actually I'll just copy-paste it into the code myself, duh.